sklearn/metrics/base.py

```diff
@@ -131,3 +131,64 @@ def _average_binary_score(binary_metric, y_true, y_score, average,
         return np.average(score, weights=average_weight)
     else:
         return score
+
+
+def _average_multiclass_ovo_score(binary_metric, y_true, y_score, average):
+    """Uses the binary metric for one-vs-one multiclass classification,
+    where the score is computed according to the Hand & Till (2001) algorithm.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples]
+        True multiclass labels.
+        Currently only handles labels with values 0 to n_classes - 1.
+
+    y_score : array, shape = [n_samples, n_classes]
+        Target scores corresponding to probability estimates of a sample
+        belonging to a particular class
+
+    average : string, ['macro' (default), 'weighted']
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean. This does not take label imbalance into account. Classes
+            are assumed to be uniformly distributed.
+        ``'weighted'``:
+            Calculate metrics for each label, taking into account the a priori
+            distribution of the classes.
+
+    binary_metric : callable, the binary metric function to use.
+        Accepts the following as input
+            y_true' : array, shape = [n_samples']
+                Some sub-array of y_true
+            y_score' : array, shape = [n_samples']
+                Target scores corresponding to the probability estimates
+                of a sample belonging to the designated positive class label
+
+    Returns
+    -------
+    score : float
+        Average the sum of the pairwise binary metric scores
+    """
+    n_labels = len(np.unique(y_true))
+    auc_scores_sum = 0
+    for pos in range(n_labels):
+        for neg in range(n_labels):
+            if pos == neg:
+                continue
+            ix = np.in1d(y_true.ravel(), [pos, neg])
+            y_true_filtered = y_true[np.where(ix.reshape(y_true.shape))]
+            y_score_filtered = y_score[np.where(ix)]
+
+            y_true_10 = y_true_filtered == pos
```

Comment:
maybe add a comment like "compute score with ``pos`` as positive class" and then "compute score with ``neg`` as positive class"?
Also, maybe then ``classa`` and classb`` or something would be better naming?